cloudify_cli/commands/agents.py

```diff
@@ -0,0 +1,137 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import threading
+
+from cloudify_cli import utils
+from cloudify_cli.logger import get_logger
+from cloudify_cli.execution_events_fetcher import wait_for_execution
+from cloudify_cli.exceptions import SuppressedCloudifyCliError
+from cloudify_cli.exceptions import ExecutionTimeoutError
+from cloudify import logs
+
+_NODE_INSTANCE_STATE_STARTED = 'started'
+
+
+def _is_deployment_installed(client, deployment_id):
+    for node_instance in client.node_instances.list(deployment_id):
+        if node_instance.state != _NODE_INSTANCE_STATE_STARTED:
+            return False
+    return True
+
+
+def _deployment_exists(client, deployment_id):
+    try:
+        client.deployments.get(deployment_id)
+    except:
+        return False
+    return True
+
+
+def install(deployment_id, include_logs):
+    workflow_id = 'install_new_agents'
+    logger = get_logger()
+    client = utils.get_rest_client()
+
+    if deployment_id:
+        deps = [deployment_id]
+        if not _deployment_exists(client, deployment_id):
+            logger.info("Deployment '{0}' does not exist"
```

Comment:
if we do need to be more explicit but there's no way for us to currently identify whether we failed to get the deployments or that they simply don't exist, maybe the message should say: `"Could not retrieve deployment for deployment_id: {0}.".format(deployment_id)`